qiskit/transpiler/target.py

```diff
@@ -0,0 +1,652 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+A target object represents the minimum set of information the transpiler needs
+from a backend
+"""
+
+from collections.abc import Mapping
+from collections import defaultdict
+import io
+import logging
+from typing import Union
+
+import retworkx as rx
+
+from qiskit.pulse.instruction_schedule_map import InstructionScheduleMap
+from qiskit.transpiler.coupling import CouplingMap
+from qiskit.transpiler.instruction_durations import InstructionDurations
+from qiskit.transpiler.timing_constraints import TimingConstraints
+
+logger = logging.getLogger(__name__)
+
+
+class InstructionProperties:
+    """A representation of the properties of a gate implementation.
+
+    This class provides the optional properties that a backend can provide
+    about an instruction. These represent the set that the transpiler can
+    currently work with if present. However, if your backend provides additional
+    properties for instructions you should subclass this to add additional
+    custom attributes for those custom/additional properties by the backend.
+    """
+
+    __slots__ = ("length", "error", "calibration")
+
+    def __init__(
+        self,
+        length: float = None,
```

Comment:
Sure, I was on the fence about duration vs length when I first added the class and went with length because of the character count. But it's better to be consistent and make this change now before we merge this and start using it everywhere

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/d27df8c6b64852d93efd829f19fc0bed417f1ffc although I left the `min_length` timing constraint parameter the same. But I can change that too if you think that makes sense.